samples/Mvc.FileUpload/Controllers/FileController.cs

```diff
@@ -56,23 +54,17 @@ public IActionResult FormToUpload()
             var formAccumulator = new KeyValueAccumulator();
             string targetFilePath = null;
 
-            var boundary = MultipartRequestHelper.GetBoundary(
-                MediaTypeHeaderValue.Parse(Request.ContentType), _defaultFormOptions.MultipartBoundaryLengthLimit);
+            var boundary = requestMediaType.GetBoundary(FormOptions.DefaultMultipartBoundaryLengthLimit);
             var reader = new MultipartReader(boundary, HttpContext.Request.Body);
 
             var section = await reader.ReadNextSectionAsync();
             while (section != null)
             {
-                ContentDispositionHeaderValue contentDisposition;
-                ContentDispositionHeaderValue.TryParse(section.ContentDisposition, out contentDisposition);
-
-                if (MultipartRequestHelper.HasFileContentDisposition(contentDisposition))
+                var fileContentSection = FileContentMultipartSection.CreateFromMultipartSection(section);
```

Comment:
This improves the user experience. BTW can the multipart reader do the check for filecontent and instead return an instance of `FileContentMultipartSection`? (that would mean this section inheriting from MulitpartSection)